### PR TITLE
feat: add storage_billing_model and default_partition_expiration_ms

### PR DESCRIPTION
**Pull request by SamuZad**
This closes https://github.com/terraform-google-modules/terraform-google-bigquery/issues/235 (which previously had a PR that was cancelled) and https://github.com/terraform-google-modules/terraform-google-bigquery/issues/333

Adds the option to specify `storage_billing_model`
Also adds in validation for `storage_billing_model`
Adds the option to specify `default_partition_expiration_ms`

Regarding validation, I've wanted to preserve the default behaviour of the underlying terraform resource so the default of `storage_billing_model` is `null`. This is to make the addition of this variable as seamless as possible. However, this means that I can't use the [contains function](https://developer.hashicorp.com/terraform/language/functions/contains) to validate the variable, as `contains()` errors out if what it checks is null

### CODE REVIEW

**Comment by SamuZad** [comment]
@davenportjw @shanecglass any chance we could get a review for this?

File: variables.tf [hunk 1]
```diff
@@ -55,12 +55,29 @@ variable "default_table_expiration_ms" {
   default     = null
 }
 
+variable "default_partition_expiration_ms" {
+  description = "The default partition expiration for all partitioned tables in the dataset, in MS"
```
Comment (g-awmalik) [inline variables.tf:59]:
```suggestion
  description = "The default partition expiration for all partitioned tables in the dataset, in msec"
```

**Review by g-awmalik — APPROVED** [review]
Thanks @SamuZad. Please address a minor nit. Looks good overall.

**Comment by SamuZad** [comment]
@g-awmalik thank you! Made the suggested change

**Comment by SamuZad** [comment]
@g-awmalik any chance for a merge?

**Comment by zckrs** [comment]
any news about this PR ?

**Comment by SamuZad** [comment]
Hello, could this PR be finally merged? It was already approved 😭

**Comment by SamuZad** [comment]
!stalebot reopen

**Comment by apeabody** [comment]
Hi @SamuZad - Could you re-push the branch to re-open the PR? Thanks!

**Comment by SamuZad** [comment]
@apeabody - I can't seem to re-open the PR despite pushing to the branch. It's fine though, I've cleaned things up a bit and created #367.

Would very much appreciate an approve and merge 🙏 Thanks in advance!

**Pull request closed without merging**